src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_params.py

```diff
@@ -0,0 +1,153 @@
+            help='Management group name.')
+        c.argument(
+            'subscription_id',
+            options_list=('--subscription-id', '-s'),
+            help='Subscription id.')
+        c.argument(
+            'resource_group_name',
+            options_list=('--resource-group-name', '-g'),
+            arg_type=resource_group_name_type,
+            completer=get_resource_group_completion_list,
+            help='Resource group name.')
+        c.argument(
+            'resource_id',
+            options_list=('--resource-id', '-r'),
+            help='Resource id.')
+        c.argument(
+            'policy_set_definition_name',
+            options_list=('--policy-set-definition-name', '-ps'),
+            completer=get_policy_set_completion_list,
+            help='Policy set definition name.')
+        c.argument(
+            'policy_definition_name',
+            options_list=('--policy-definition-name', '-pd'),
+            completer=get_policy_completion_list,
+            help='Policy definition name.')
+        c.argument(
+            'policy_assignment_name',
+            options_list=('--policy-assignment-name', '-pa'),
+            completer=get_policy_assignment_completion_list,
+            help='Policy assignment name.')
+        c.argument(
+            'from_value',
+            options_list=('--from'),
+            help='Timestamp specifying the start time of the interval to query.')
+        c.argument(
+            'to_value',
+            options_list=('--to'),
+            help='Timestamp specifying the end time of the interval to query.')
+        c.argument(
+            'top_value',
+            options_list=('--top'),
+            type=int,
+            help='Maximum number of records to return.')
+        c.argument(
+            'order_by_clause',
+            options_list=('--order-by'),
+            help='Ordering expression using OData notation.')
+        c.argument(
+            'select_clause',
+            options_list=('--select'),
+            help='Select expression using OData notation.')
+        c.argument(
+            'filter_clause',
+            options_list=('--filter'),
+            help='Filter expression using OData notation.')
+        c.argument(
+            'apply_clause',
+            options_list=('--apply'),
+            help='Apply expression for aggregations using OData notation.')
+
+    with self.argument_context('policy state') as c:
+        c.argument(
+            'all_results',
+            options_list=('--all'),
+            arg_type=get_three_state_flag(),
+            help='Within the specified time interval, get all policy states instead of the latest only.')
+        c.argument(
+            'management_group_name',
+            options_list=('--management-group-name', '-m'),
+            help='Management group name.')
+        c.argument(
+            'subscription_id',
+            options_list=('--subscription-id', '-s'),
```

Comment:
Not sure I understand the question. Current subscription concept is local to PS/CLI abstractions. As far as ARM is concerned, an extension RP can be called with any subscription. PS/CLI uses a "current" subscription concept so that users don't need to specify it over an over again. Same as here, in PS I made subscriptionid an optional argument, so that when missing it uses the "current" but customers can specify it without having to switch the current.

Reply:
This will potentially cause problems when we introduce `--subscription` as a global argument (which is in progress). My preference would simply be not to expose it for now and instead wait until this global parameter is exposed for all commands.

Reply:
Oh, ok. Sounds good. I'll remove --subscription-id parameter.